docs/topics/Threads.md

```diff
@@ -0,0 +1,111 @@
+- The [Thread List Sync](#DOCS_TOPICS_GATEWAY/thread-list-sync) Gateway Event will contain an array of [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) objects for any returned threads the current user is a member of.
+
+### Syncing for other users
+
+> info
+> These require the `GUILD_MEMBERS` [Gateway Intent](#DOCS_TOPICS_GATEWAY/gateway-intents)
+
+- An API `GET` call to `/channels/<channel_id>/thread-members` which returns an array of [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) objects.
+- The [Thread Members Update](#DOCS_TOPICS_GATEWAY/thread-members-update) Gateway Event which will include all users who were added to or removed from a thread by an action.
+
+## Editing & Deleting Threads
+
+Threads can be edited and deleted with the existing `PATCH` and `DELETE` endpoints to edit a channel.
+
+- Deleting a thread requires the `MANAGE_THREADS` permission.
+- Editing a thread to set `archived` to `false` only requires the current user has already been added to the thread. If `locked` is true, then the user must have `MANAGE_THREADS`
+- Editing a thread to change the `name`, `archived`, `auto_archive_duration` fields requires `MANAGE_THREADS` or that the current user is the thread creator
+- Editing a thread to change `rate_limit_per_user` or `locked` requires `MANAGE_THREADS`
+
+## NSFW Threads
+
+Threads do not explicitly set the `nsfw` field. All threads in a channel marked as `nsfw` inherit that setting though.
+
+## New Message Types
+
+Threads introduce a few new [message types](#DOCS_RESOURCES_CHANNEL/message-object-message-types), and re-purpose some others:
+
+- `RECIPIENT_ADD` and `RECIPIENT_REMOVE` have been repurposed to also send when a user is added to or removed from a thread by someone else
+- `CHANNEL_NAME_CHANGE` has been repurposed and is sent when the thread's name is changed
+- `THREAD_CREATED` is a new message sent when a thread is created from an older message, it contains a [message reference](#DOCS_RESOURCES_CHANNEL/message-reference-object-message-reference-structure) with the `guild_id` and `channel_id` of the thread. The `content` of the message is the `name` of the thread.
+- `THREAD_STARTER_MESSAGE` is a new message sent as the first message in certain threads. It _only_ contains a [message reference](#DOCS_RESOURCES_CHANNEL/message-reference-object-message-reference-structure) field that points to the message from which the thread was started.
+
+## Enumerating threads
+
+There are four new `GET` routes for enumerating threads in a specific channel
+
+- `/channels/<channel_id>/threads/active` returns all active threads in a channel that the current user can access, includes public & private threads
+- `/channels/<channel_id>/users/@me/threads/archived/private` returns all archived, private threads in a channel, that the current user has is a member of, sorted by thread id descending
```

Comment:
Noticing that there isn't a `public` analogous version of 
`/channels/<channel_id>/users/@me/threads/archived/private`

There is `/channels/<channel_id>/threads/archived/public` but it isn't quite the same.

Is there a reason that we can't enumerate public archived threads that we are a member of?

Reply:
I think it's just that the public one will already include all of those, so there's no need for it.

For private threads, it becomes more of an issue because of the need to do permission checking but also maintain efficient pagination in our database, so that's why it's split into two routes.

Reply:
So the idea here is that if you want the functionality of `/channels/<channel_id>/users/@me/threads/archived/public` you should call `/channels/<channel_id>/threads/archived/public` and filter based on whether you have a `thread.member` for the thread?

Reply:
Yeah, my assumption was that the number of active threads is much much smaller than the number of archived threads so we don't need any pagination there, and that makes it easy to just do the filtering in your code?  (That's what our clients do at least, they load all active threads and then filter client side)